### PR TITLE
move sources to winning message and encrypt it

### PR DESCRIPTION
**Pull request by merlinran**

### CODE REVIEW

File: .github/workflows/review.yml [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
+
 on:
   push:
     branches:
       - main
   pull_request:
+
+concurrency:
+  group: test-${{github.ref}}
+  cancel-in-progress: true
```
Comment (merlinran) [inline .github/workflows/review.yml:11]: cancel concurrent CI runs just in case, to save some resources.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -1,11 +1,15 @@
+name: Test
+
 on:
   push:
     branches:
       - main
   pull_request:
-    branches:
-      - main
```
Comment (merlinran) [inline .github/workflows/test.yml:7]: Interesting this was only for main branch.

File: service/service.go [hunk 3]
```diff
@@ -503,28 +500,50 @@ func (s *Service) filterAuction(auction *pb.Auction) (rejectReason string) {
...
 	if err != nil {
-		log.Errorf("error getting bid, assuming bytes limit is not hit: %v", err)
-	} else {
-		// request for some quota, which may be used or gets expired if lossing
-		// the auction.
-		granted := s.bytesLimiter.Request(win.AuctionId, bid.DealSize, BidsExpiration)
-		if !granted {
-			return nil, errWouldExceedRunningBytesLimit
-		}
+		return nil, fmt.Errorf("getting bid: %v", err)
```
Comment (merlinran) [inline service/service.go:510]: fail in this case, because we need its payload CID now.

File: service/store/store.go [hunk 4]
```diff
@@ -246,9 +246,6 @@ func validate(b Bid) error {
 	if err := b.AuctioneerID.Validate(); err != nil {
 		return fmt.Errorf("auctioneer id is not a valid peer id: %v", err)
 	}
-	if err := b.Sources.Validate(); err != nil {
```
Comment (merlinran) [inline service/store/store.go:249]: No longer validate Sources here.

File: .github/workflows/test.yml (hunk 2 above)
Comment (jsign) [reply .github/workflows/test.yml:7]: Weird.

File: proto/v1/message.proto [hunk 5]
```diff
@@ -28,6 +28,7 @@ message Bid {
 message WinningBid {
     string auction_id = 1;
     string bid_id = 2;
+    bytes encrypted_sources = 3; // Sources message encrypted by bidbot's public key.
```
[the remark this answers is not in the document]
Comment (jsign) [reply proto/v1/message.proto:31]: Maybe we can rename to `encrypted_metadata` or something more generic?
I'm thinking if we might end up adding more stuff in the encrypted field.

So mostly an "encrypted protobuf that contains extra private information to share with winners". For now would only be "Sources" data, but maybe in the future there might be more things that we can evolve that protobuf to contain in a backwards compat way. That means that the encrypted thing would be a new proto message, which now the only field is `pb.Sources`.

Other option is not doing this, and just keep adding new "encryptedField". But that might be a bit verbose and multiple encryption/decryption work.

What do you think?

File: service/store/store.go (hunk 4 above)
Comment (jsign) [reply service/store/store.go:249]: If this validation isn't here, should have a new place somewhere else in the code? I think might be in `SetAwaitingProposalCid` where gets persisted?

**Review by jsign — APPROVED** [review]
:100:  
Left some comments for your consideration.

File: proto/v1/message.proto (hunk 5 above)
[the remark this answers is not in the document]
Comment (merlinran) [reply proto/v1/message.proto:31]: Good idea. Maybe just `encrypted`.

File: service/store/store.go (hunk 4 above)
Comment (merlinran) [reply service/store/store.go:249]: The caller of `SetAwaitingProposalCid` in service.go does validation, but no harm to be more cautious. Will add

File: proto/v1/message.proto (hunk 5 above)
[the remark this answers is not in the document]
Comment (merlinran) [reply proto/v1/message.proto]: e2746d0496dd16ee0b7cc78529c897a721d442d8

**Pull request merged**